GeneralsMD/Code/GameEngineDevice/Source/W3DDevice/GameClient/W3DView.cpp

```diff
@@ -1538,6 +1538,8 @@ void W3DView::draw( void )
 	CustomScenePassModes customScenePassMode  = SCENE_PASS_DEFAULT;
 	Bool preRenderResult = false;
 
+	if (TheGlobalData->m_headless)
```

Comment:
Shouldn't this test be higher up near
```cpp
	if(TheGlobalData->m_playIntro || TheGlobalData->m_afterIntro)
	{
		// redraw all views, update the GUI
		{
			TheDisplay->DRAW();
		}
		{
			TheDisplay->UPDATE();
		}
		return;
	}
```

and

```cpp
	// update display
	{
		TheDisplay->UPDATE();
	}

	{
		USE_PERF_TIMER(GameClient_draw)
			
	// redraw all views, update the GUI
	//if(TheGameLogic->getFrame() >= 2)
		
		TheDisplay->DRAW();
	}
```
?

Same with W3DDisplay::draw